presto-main/src/test/java/io/prestosql/server/ui/TestWebUi.java

```diff
@@ -344,13 +344,115 @@ private static void assertRedirect(OkHttpClient client, String url, String redir
+            try (Response response = client.newCall(request).execute()) {
+                assertEquals(response.code(), SC_SEE_OTHER);
+                assertEquals(
+                        response.header(LOCATION),
+                        uriBuilderFrom(URI.create(redirectLocation))
+                                .scheme("test")
+                                .port(123)
+                                .toString());
+            }
+
+            request = new Request.Builder()
+                    .url(url)
+                    .header(X_FORWARDED_PROTO, "test")
+                    .build();
+            try (Response response = client.newCall(request).execute()) {
+                assertEquals(response.code(), SC_SEE_OTHER);
+                assertEquals(
+                        response.header(LOCATION),
+                        uriBuilderFrom(URI.create(redirectLocation))
+                                .scheme("test")
+                                .defaultPort()
+                                .toString());
+            }
+            request = new Request.Builder()
+                    .url(url)
+                    .header(X_FORWARDED_HOST, "my-load-balancer.local")
+                    .header(X_FORWARDED_PORT, "123")
+                    .build();
+            try (Response response = client.newCall(request).execute()) {
+                assertEquals(response.code(), SC_SEE_OTHER);
+                assertEquals(
+                        response.header(LOCATION),
+                        uriBuilderFrom(URI.create(redirectLocation))
+                                .host("my-load-balancer.local")
+                                .port(123)
+                                .toString());
+            }
+
+            request = new Request.Builder()
+                    .url(url)
+                    .header(X_FORWARDED_HOST, "my-load-balancer.local:123")
+                    .build();
+            try (Response response = client.newCall(request).execute()) {
+                assertEquals(response.code(), SC_SEE_OTHER);
+                assertEquals(
+                        response.header(LOCATION),
+                        uriBuilderFrom(URI.create(redirectLocation))
+                                .host("my-load-balancer.local")
+                                .port(123)
+                                .toString());
+            }
+
+            request = new Request.Builder()
+                    .url(url)
+                    .header(X_FORWARDED_HOST, "my-load-balancer.local")
+                    .build();
+            try (Response response = client.newCall(request).execute()) {
+                assertEquals(response.code(), SC_SEE_OTHER);
+                assertEquals(
+                        response.header(LOCATION),
+                        uriBuilderFrom(URI.create(redirectLocation))
+                                .host("my-load-balancer.local")
+                                .defaultPort()
```

Comment:
Isn't in this case simply `http://my-load-balancer.local/`? we could use constant here

Reply:
No. The code is `uriBuilderFrom`, so the subsequent lines are "replacements".